ambry-server/src/test/java/com.github.ambry.server/AmbryRequestsTest.java

```diff
@@ -330,6 +331,81 @@ public void catchupStatusFailureTest() throws InterruptedException, IOException
     doCatchupStatusTest(null, 0, Short.MAX_VALUE, ServerErrorCode.Unknown_Error, false);
   }
 
+  /**
+   * Tests for the response received on a {@link StopBlobStoreAdminRequest} for successful case
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  @Test
+  public void stopBlobStoreSuccessTest() throws InterruptedException, IOException {
+    List<? extends PartitionId> partitionIds = clusterMap.getAllPartitionIds();
+    assertTrue("This test needs more than one partition to work", partitionIds.size() > 1);
+    PartitionId id = partitionIds.get(0);
+    List<? extends ReplicaId> replicaIds = id.getReplicaIds();
+    assertTrue("This test needs more than one replica for the first partition to work", replicaIds.size() > 1);
+    int correlationId = TestUtils.RANDOM.nextInt();
+    String clientId = UtilsTest.getRandomString(10);
+    long acceptableLagInBytes = 0;
+    short numReplicasCaughtUpPerPartition = 3;
+    replicationManager.reset();
+    replicationManager.controlReplicationReturnVal = true;
+    generateLagOverrides(0, acceptableLagInBytes);
+    AdminRequest adminRequest = new AdminRequest(AdminRequestOrResponseType.StopBlobStore, id, correlationId, clientId);
+    StopBlobStoreAdminRequest stopBlobStoreAdminRequest =
+        new StopBlobStoreAdminRequest(numReplicasCaughtUpPerPartition, adminRequest);
+    Response response = sendRequestGetResponse(stopBlobStoreAdminRequest, ServerErrorCode.No_Error);
+    assertTrue("Response not of type AdminResponse", response instanceof AdminResponse);
```

Comment:
you should make sure all the apis have been called

Reply:
APIs are verified in the following code snippet.